src/abi.md

```diff
@@ -1,21 +1,209 @@
 # Application Binary Interface (ABI)
 
-This section documents features that affect the ABI of the compiled output of
-a crate.
+r[abi]
 
-See *[extern functions]* for information on specifying the ABI for exporting
-functions. See *[external blocks]* for information on specifying the ABI for
-linking external libraries.
+## ABI Compatibility
+
+r[abi.compatibility]
```

Comment:
Can you add a few sentences explaining what *abi compatibility* is? Such as the consequence of what it means for types to be abi compatible? In general, I felt a little uncertain what I was reading, since it immediately jumps into a list of rules without giving context or explaining what this section is about.